src/api/app/models/group.rb

```diff
@@ -208,6 +208,7 @@ def maintainer_roler
 
   def delete_user(klass, login_id, group_id)
     klass.where('user_id = ? AND group_id = ?', login_id, group_id).delete_all if [GroupMaintainer, GroupsUser].include?(klass)
+    Event::RemovedUserFromGroup.create(group: Group.find(group_id).title, user: User.find(login_id).login, who: User.session&.login) if klass == GroupsUser
```

Comment:
@saraycp hmm, leaking the user session into the models is not great. I know we have many places where this is happening. But would be great if we could avoid this.

Reply:
:thinking: thinking of alternatives...

Reply:
@saraycp alright, yeah, since this is always handled slightly different (API, webui, different methods) we would have to add a considerable amount of more code changes in order to avoid this. Its not nice, but dont want to block this from going further. Its maybe something for another refactoring. I will create a card to track it

Reply:
We agreed on postponing this. It will require some refactoring. It's tracked on this [Trello card](https://trello.com/c/nw9Z76QI/183-stop-using-usersession-in-models)